src/EditorFeatures/CSharpTest/MoveDeclarationNearReference/MoveDeclarationNearReferenceTests.cs

```diff
@@ -1326,6 +1326,71 @@ void Local<T>()
     }
 
     public static void Out<T>(out T t) => t = default;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMoveInsideSwitchSection()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        switch (true)
+        {
+            case true:
+                int [||]x = 0;
+                System.Console.WriteLine();
+                System.Console.WriteLine(x);
+                break;
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        switch (true)
+        {
+            case true:
+                System.Console.WriteLine();
+                int x = 0;
+                System.Console.WriteLine(x);
+                break;
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMoveIntoSwitchSection()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        int [||]x;
+        switch (true)
+        {
+            case true:
+                x = 0;
+                break;
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        switch (true)
+        {
+            case true:
+                int x = 0;
+                break;
+        }
+    }
 }");
         }
```

Comment:
can we have a test where it's used from multiple switch sections?

Reply:
➡️ Added in d49568c966a0404f7979ce55f979093a2c5ba86a